Packs/QRadar/Integrations/QRadar_v3/QRadar_v3.py

```diff
@@ -1642,22 +1638,25 @@ def enrich_offense_with_events(client: Client, offense: Dict, fetch_mode: str, e
     events_count = offense.get('event_count', 0)
     events: List[dict] = []
     failure_message = ''
-    search_id = create_search_with_retry(client, fetch_mode, offense, events_columns,
-                                         events_limit)
     is_success = True
-    if search_id == QueryStatus.ERROR.value:
-        failure_message = 'Search for events was failed.'
-        is_success = False
-    else:
-        events, failure_message = poll_offense_events_with_retry(client, search_id, int(offense_id))
-    events_fetched = sum(int(event.get('eventcount', 1)) for event in events)
-    print_debug_msg(f'Events fetched for offense {offense_id}: {events_fetched}/{events_count}.')
-    offense['events_fetched'] = events_fetched
-    if events:
-        offense['events'] = events
+    for _ in range(MAX_FETCH_EVENT_RETIRES):
+        search_id = create_search_with_retry(client, fetch_mode, offense, events_columns,
+                                             events_limit)
+        if search_id == QueryStatus.ERROR.value:
+            failure_message = 'Search for events was failed.'
+        else:
+            events, failure_message = poll_offense_events_with_retry(client, search_id, int(offense_id))
+        events_fetched = sum(int(event.get('eventcount', 1)) for event in events)
+        print_debug_msg(f'Events fetched for offense {offense_id}: {events_fetched}/{events_count}.')
+        offense['events_fetched'] = events_fetched
+        if events:
+            offense['events'] = events
+            break
+        print_debug_msg(f'No events were fetched for offense {offense_id}. Retrying in {FAILURE_SLEEP}.')
+        time.sleep(FAILURE_SLEEP)
     else:
-        print_debug_msg(f'No events were fetched for offense {offense_id}'
-                        f'Adding to mirroring queue to be queried again.')
+        print_debug_msg(f'No events were fetched for offense {offense_id}. '
+                        f'If mirroring is enabled, it will be queried again in mirroring.')
```

Comment:
What if mirroring isn't enabled?
Should we advise the user to enable mirroring?

Reply:
If mirroring isn't enabled, the events will not be fetched at all. We already advise enabling mirroring in this case